### ISSUE TITLE
Discrepancy in Results for iBVP_PhysNet on PURE Dataset

### ISSUE THREAD

**Issue by alaahejer** [PhysiologicAILab/FactorizePhys#3]
Dear authors,

I have been testing the model `iBVP_PhysNet.pth` using the configuration file `iBVP_PURE_PHYSNET_BASIC.yaml` on the full PURE dataset (58 folders). However, I observed that the results I obtained during testing significantly differ from the results presented in your paper, as highlighted in the red box in the table. 

Specifically, the metrics I obtained for the PURE dataset testing do not align with the reported values in the paper. Additionally, the signal extracted during the model evaluation on this dataset is as follows:  

<img width="334" alt="Image" src="https://github.com/user-attachments/assets/16059d78-9825-43cb-9fae-641ab667bda7" />

Could you kindly clarify if there are any additional preprocessing steps, parameter adjustments, or dataset-specific considerations that need to be applied for testing? Your insights would be greatly appreciated.

Thank you for your attention to this matter. 

Best regards,  
Alaa Hajr

<img width="468" alt="Image" src="https://github.com/user-attachments/assets/08ae11be-c6f2-4789-a084-6920f84510f4" />

**Comment by jnj256** [comment]
Dear Alaa,
We report the preprocessing steps in the paper - in supplementary section, which should be helpful. If you can share some additional details as follows, I'll be able to support you better:
- Data preprocessing steps followed/ Config file that you used to prepare PURE dataset
- Performance metrics that you obtained when testing the model on PURE dataset.

**Comment by alaahejer** [comment]
Dear authors,

I obtained the following results, and there is a difference between my results and the ones reported in your paper.

Despite using the full PURE dataset to test the model iBVP-PhysFormer-PURE, and applying the exact same configuration file settings provided on your GitHub repository, some of the evaluation metrics did not match accurately. However, some results were within close range of the values reported in your paper and on the website.

![Image](https://github.com/user-attachments/assets/75e3dfa7-2a9e-4401-8ebd-02d1b7835bf5)

Best regards and warm greetings.

**Comment by jnj256** [comment]
[PURE_Data_list.txt](https://github.com/user-attachments/files/18690890/PURE_Data_list.txt)

Can you please verify that after processing PURE dataset, you get exactly the same files as listed in the attached list. If this is not the case, please share the config file.

**Comment by GSW2221** [comment]
Dear Author, I read the Implementation Overview section 11 of your paper, which mentioned that FactorizePhys utilizes the same preprocessing method (Diff) as PhysNet and PhysFormer. But the configuration file in your open source link is set to Raw, does that mean the user has to adjust it manually? I saw in @alaahejer's answer that he also used the exact same profile on GitHub, maybe his error had something to do with that.

**Comment by jnj256** [comment]
> Dear Author, I read the Implementation Overview section 11 of your paper, which mentioned that FactorizePhys utilizes the same preprocessing method (Diff) as PhysNet and PhysFormer. But the configuration file in your open source link is set to Raw, does that mean the user has to adjust it manually? I saw in [@alaahejer](https://github.com/alaahejer)'s answer that he also used the exact same profile on GitHub, maybe his error had something to do with that.

@GSW2221 : Thank you, for your observation on this. FactorizePhys and all models in this repo - use Diff as a preprocessing step. Diff is implemented within the model implementation as it can be seen here: https://github.com/PhysiologicAILab/FactorizePhys/blob/de72e58e5f83065f14609f7f11403d7294e99127/neural_methods/model/FactorizePhys/FactorizePhys.py#L205 

The input to FactorizePhys and other models in this repo - is therefore to be specified as `Raw'. Let me know if you encounter any specific issues or have any other observations in this.

**Comment by jnj256** [comment]
@alaahejer : Would you like to share further updates about the discrepancy of results for PhysNet model (trained on iBVP dataset, evaluated on PURE)? 
If PURE dataset was prepared using other toolbox (r.g. rPPG-Toolbox), it is quite possible that this issue may relate to the face-detector used to prepare the dataset. Earlier version of rPPG-Toolbox, for instance, provided support for RetinaFace and Haar Cascade face-detector. This repo uses YOLO5Face - which was found to perform quite well than others. The most recent update of rPPG-Toolbox repo - added the support for YOLO5Face. If the dataset was prepared earlier, it may be worth re-preparing the dataset and then perform the evaluation.

**Comment by GSW2221** [comment]
@jnj256 Thank you very much for your reply. In addition, I still have some doubts about LABEL_TYPE: I observed from the library that there are two types of label processing Standardized and DiffNormalized, but the selection of LABEL_TYPE is not mentioned in many papers. I was reading your paper when I saw the relevant description on page 7, section 4: "For fair evaluation, all model-specific hyperparameters were maintained as provided by the respective SOTA rPPG methods, while the training pipeline related hyperparameters, which include preprocessing steps for images and labels, batch size, number of epochs, learning rate, scheduler, and optimizer were kept consistent for training all the models." 

May I ask why the label processing methods of all the comparison algorithms were uniformly Standardized when you did the comparison experiment? Why not select DiffNormalized?The reason I bring this up is because in rPPG-ToolBox, PhysNet's LABEL_TYPE is set to DiffNormalized. And in your paper it becomes Standardized.Looking forward to your reply!

**Comment by jnj256** [comment]
> May I ask why the label processing methods of all the comparison algorithms were uniformly Standardized when you did the comparison experiment? Why not select DiffNormalized?The reason I bring this up is because in rPPG-ToolBox, PhysNet's LABEL_TYPE is set to DiffNormalized. And in your paper it becomes Standardized.Looking forward to your reply!

@GSW2221 : That's an excellent observation. We would like to draw your attention to second para of A.2 in the supplementary section on page 16. It actually relates to your query, and this is what it states: 
> Ground-truth BVP signals are also uniformly standardized for training all models. This is different from some of the recent work [37 ] that applies Diff in addition to standardization. We empirically found that all models perform significantly better when trained with the standardized BVP signals, although when the Diff is applied to the video frames.

Just to also clarify, 
As initially we found most SOTA performing quite well using only standardized (not DiffNormalized) ground-truth PPG signals, we trained all models from scratch in this manner.

Hope this addresses your query. Feel free to share your further thoughts on this.

**Comment by GSW2221** [comment]
> > Hope this addresses your query. Feel free to share your further thoughts on this.

Thank you very much for your reply. Both your reply and your paper have greatly inspired and helped me.

**Comment by jnj256** [comment]
Hi @alaahejer,
As I've not heard from you recently, closing this issue, assuming it was resolved at your end. Feel free to reopen and share further findings if you still observe the discrepancy.
